### PR TITLE
feat: add BasicAbility dcc client and adjust dcc configuration directory

### PR DESCRIPTION
**Pull request by codding-y**
# Description

Add BasicAbility dcc client and adjust dcc configuration directory

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/ConfigurationApiClient.cs [hunk 1]
```diff
@@ -136,6 +136,37 @@ protected virtual (string Raw, ConfigurationTypes ConfigurationType) FormatRaw(s
...
+                    throw new ArgumentException("configObject invalid");
+                }
+
+            case ConfigFormats.Yaml:
+                try
+                {
+                    var deserializer = new DeserializerBuilder().Build();
+                    var yamlObject = deserializer.Deserialize<object>(result.Content!);
+
+                    var serializer = new SerializerBuilder().JsonCompatible().Build();
```
Comment (zhenlei520) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/ConfigurationApiClient.cs:159]: No need to initialize every time

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Masa.Utils.Caching.DistributedMemory" Version="$(MasaUtilsPackageVersion)" />
+    <PackageReference Include="Masa.Utils.Caching.Redis" Version="$(MasaUtilsPackageVersion)" />
+    <PackageReference Include="Masa.Utils.Caller.HttpClient" Version="$(MasaUtilsPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="$(MicrosoftPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="$(MicrosoftPackageVersion)" />
+    <PackageReference Include="Microsoft.Extensions.Http" Version="$(MicrosoftPackageVersion)" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (zhenlei520) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj:16]: Must use `Newtonsoft.Json`

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    /// <summary>
+    /// The cluster name.
+    /// </summary>
+    public string? Cluster { get; set; }
+
+    /// <summary>
+    /// The app id.
+    /// </summary>
+    public string AppId { get; set; } = default!;
```
Comment (zhenlei520) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.md:87]: Does the local node need an AppId、ConfigObjects 、Secret 、Cluster ?

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    public List<string> ConfigObjects { get; set; } = default!;
+
+    public string? Secret { get; set; }
+
+    /// <summary>
+    /// 将CustomDccSectionOptions挂载到根节点下
+    /// </summary>
+    [JsonIgnore]
+    public virtual string? Section => string.Empty;
+}
```
Comment (zhenlei520) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:101]: same as above

File: test/Masa.Contrib.BasicAbility.Dcc.Tests/Masa.Contrib.BasicAbility.Dcc.Tests/Masa.Contrib.BasicAbility.Dcc.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    <None Remove="appsettings.json" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="appsettings.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (zhenlei520) [inline test/Masa.Contrib.BasicAbility.Dcc.Tests/Masa.Contrib.BasicAbility.Dcc.Tests/Masa.Contrib.BasicAbility.Dcc.Tests.csproj:21]: using `Directory.Build.props`

File: test/Masa.Contrib.BasicAbility.Pm.Tests/Masa.Contrib.BasicAbility.Pm.Tests.csproj [hunk 6]
```diff
@@ -12,7 +12,10 @@
     <PackageReference Include="Moq" Version="$(MoqPackageVersion)" />
     <PackageReference Include="MSTest.TestAdapter" Version="$(MSTestPackageVersion)" />
     <PackageReference Include="MSTest.TestFramework" Version="$(MSTestPackageVersion)" />
-    <PackageReference Include="coverlet.collector" Version="$(CoverletPackageVersion)" />
+    <PackageReference Include="coverlet.collector" Version="3.1.2">
+      <PrivateAssets>all</PrivateAssets>
```
Comment (zhenlei520) [inline test/Masa.Contrib.BasicAbility.Pm.Tests/Masa.Contrib.BasicAbility.Pm.Tests.csproj:16]: using `Directory.Build.props`

File: src/BasicAbility/Masa.Contrib.BasicAbility.Dcc/Masa.Contrib.BasicAbility.Dcc.csproj [hunk 7]
```diff
@@ -7,16 +7,13 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Masa.Utils.Caching.DistributedMemory" Version="$(MasaUtilsPackageVersion)" />
-    <PackageReference Include="Masa.Utils.Caching.Redis" Version="$(MasaUtilsPackageVersion)" />
-    <PackageReference Include="Masa.Utils.Caller.HttpClient" Version="$(MasaUtilsPackageVersion)" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="$(MicrosoftPackageVersion)" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="$(MicrosoftPackageVersion)" />
-    <PackageReference Include="Microsoft.Extensions.Http" Version="$(MicrosoftPackageVersion)" />
+    <PackageReference Include="Masa.Utils.Caching.DistributedMemory" Version="0.5.0-preview.3" />
+    <PackageReference Include="Masa.Utils.Caching.Redis" Version="0.5.0-preview.3" />
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.Contrib.BasicAbility.Dcc/Masa.Contrib.BasicAbility.Dcc.csproj:11]: using `Directory.Build.props`

File: src/BasicAbility/Masa.Contrib.BasicAbility.Dcc/README.md [hunk 8]
```diff
@@ -1,33 +1,27 @@
 [中](README.zh-CN.md) | EN
 
-## Masa.Contrib.BasicAbility.Dcc
+## Masa.Contrib.BasicAbility.Pm
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.Contrib.BasicAbility.Dcc/README.md:3]: Masa.Contrib.BasicAbility.Pm?

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj (hunk 2 above)
Comment (codding-y) [reply src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj:16]: Subsequent planning will be abstracted

**Comment by zhenlei520** [comment]
@doddgu

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj (hunk 2 above)
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Masa.Contrib.Configuration.ConfigurationApi.Dcc.csproj:16]: Newtonsoft.Json???

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    options.OnChange(option =>
+    {
+        //TODO 配置更新
+    });
+    return System.Text.Json.JsonSerializer.Serialize(option.CurrentValue);
+});
+
+app.MapGet("/GetPlatformName", ([FromServices] IConfiguration configuration) =>
+{
+    //格式：ConfigurationAPI:<Replace-With-Your-AppId>:<Your Node Path>:<parameter name>
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:127]: 从配置中心获取指定AppId下的Name的配置值

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+```
+
+```C#
+builder.AddMasaConfiguration(configurationBuilder => configurationBuilder.UseDcc());//使用Dcc提供远程配置的能力
+
+/// <summary>
+/// 自动映射节点关系
+/// </summary>
+public class PlatformOptions : ConfigurationApiMasaConfigurationOptions
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:57]: Platforms?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 11]
```diff
@@ -54,7 +54,7 @@ builder.AddMasaConfiguration(configurationBuilder => configurationBuilder.UseDcc
 /// <summary>
 /// 自动映射节点关系
 /// </summary>
-public class PlatformOptions : ConfigurationApiMasaConfigurationOptions
+public class RedisOptions : ConfigurationApiMasaConfigurationOptions
```
[the remark this answers is not in the document]
Comment (codding-y) [reply src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:57]: This is an example. Take get the redis configuration as an example

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Internal/Parser/XmlConfigurationParser.cs [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+        return DatasetToJson(dataSet);
+    }
+
+    public static DataSet XmlToDataSet(string xmlStr)
+    {
+        try
+        {
+            var ds = new DataSet();
+            using var xmlSR = new StringReader(xmlStr);
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Internal/Parser/XmlConfigurationParser.cs:23]: xmlSR -> xmlStringReader or xmlSr

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Internal/Parser/XmlConfigurationParser.cs [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+            json.Append("\":");
+            json.Append(DataTableToJson(dt));
+            json.Append('}');
+        }
+        return json.ToString();
+    }
+
+    public static string DataTableToJson(DataTable table)
+    {
+        var JsonString = new StringBuilder();
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/Internal/Parser/XmlConfigurationParser.cs:52]: JsonString -> json

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 14]
```diff
@@ -126,11 +123,11 @@ app.MapGet("/GetPlatformByMonitor", ([FromServices] IOptionsMonitor<PlatformOpti
...
 
-app.MapGet("/GetPlatformName", ([FromServices] IConfiguration configuration) =>
+app.MapGet("/GetRedisHost", ([FromServices] IConfiguration configuration) =>
 {
-    //从配置中心获取指定AppId下的指定配置对象（ConfigObject）的Name的配置值
-    //格式：ConfigurationAPI:<Replace-With-Your-AppId>:<Your ConfigObject>:<parameter name>
-    return configuration["ConfigurationAPI:<Replace-With-Your-AppId>:Platforms:Name"];
+    //从配置中心获取指定AppId下的指定配置对象（ConfigObject）的Host的配置值
+    //格式：ConfigurationAPI:<自定义AppId>:<自定义的ConfigObject>:<参数名Name>
+    return configuration["ConfigurationAPI:<Replace-With-Your-AppId>:Redis:Name"];
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:130]: Name?

File: src/Configuration/Masa.Contrib.Configuration/README.zh-CN.md [hunk 15]
```diff
@@ -108,10 +108,10 @@ app.Map("/GetPlatform", ([FromServices] IOptionsMonitor<PlatformOptions> option)
     return System.Text.Json.JsonSerializer.Serialize(option.CurrentValue);
 });//推荐（需要自动或手动映射节点关系后才能使用）
 
-app.Map("/GetPlatformName", ([FromServices] IConfiguration configuration) =>
+app.Map("/GetRedisformName", ([FromServices] IConfiguration configuration) =>
 {
     //基础
-    return configuration["Local:Platforms:Name"];
+    return configuration["Local:Redis:Name"];
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration/README.zh-CN.md:114]: Host? Name? Be the same!

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md [hunk 16]
```diff
@@ -126,8 +126,8 @@ app.MapGet("/GetRedisByMonitor", ([FromServices] IOptionsMonitor<RedisOptions> o
 app.MapGet("/GetRedisHost", ([FromServices] IConfiguration configuration) =>
 {
     //从配置中心获取指定AppId下的指定配置对象（ConfigObject）的Host的配置值
-    //格式：ConfigurationAPI:<自定义AppId>:<自定义的ConfigObject>:<参数名Name>
-    return configuration["ConfigurationAPI:<Replace-With-Your-AppId>:Redis:Name"];
+    //格式：ConfigurationAPI:<自定义AppId>:<自定义的ConfigObject>:<参数名Host>
+    return configuration["ConfigurationAPI:<Replace-With-Your-AppId>:Redis:Host"];
```
Comment (doddgu) [inline src/Configuration/Masa.Contrib.Configuration.ConfigurationApi.Dcc/README.zh-CN.md:130]: You forget to update its en version?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**